datafusion/physical-expr/src/expressions/in_list.rs

```diff
@@ -474,16 +427,46 @@ pub fn in_list(
     )))
 }
 
+/// Create a new InList expression directly from an array, bypassing expression evaluation.
+///
+/// This is more efficient than `in_list()` when you already have the list as an array,
+/// as it avoids the conversion: ArrayRef -> Vec<PhysicalExpr> -> ArrayRef -> ArraySet.
+/// Instead it goes directly: ArrayRef -> ArraySet.
+///
+/// The `list` field will be empty when using this constructor, as the array is stored
+/// directly in the static filter.
+pub fn in_list_from_array(
```

Comment:
I wonder if it would be more discoverable if this was a method on `InList` rather than a free function

Something like 
```rust
impl InLIst
  fn new_from_array(  expr: Arc<dyn PhysicalExpr>,
    array: ArrayRef,
    negated: bool,
   ) -> Result<Self> {
...
}
```

Reply:
Yeah I agree, I was just following the existing patterns